src/guides/resolve-zil-domain.md

```diff
@@ -0,0 +1,244 @@
+
+```markup
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="utf-8">
+        <title>Basic .zil integration</title>
+    </head>
+    <body>
+
+      <script 
+        src="https://cdnjs.cloudflare.com/ajax/libs/js-sha256/0.9.0/sha256.min.js" 
+        integrity="sha512-szJ5FSo9hEmXXe7b5AUVtn/WnL8a5VofnFeYC2i2z03uS2LhAch7ewNLbl5flsEmTTimMN0enBZg/3sQ+YOSzQ==" 
+        crossorigin="anonymous"></script>
+
+      <script src="index.js"></script>
+    </body>
+</html>
+```
+
+Don't forget to add an input field and a button that will trigger the resolution process
+
+```markup
+      <input id="input" />
+      <button onclick="resolve()">Resolve</button>
+```
+
+#### As for our index.js file
+
+We are going to put some basic code to capture the text from the input field and print it in our console
+
+```javascript
+// <!-------> Resolving domain <!------->
+async function resolve() {
+  const userInput = document.getElementById("input").value;
+  console.log({ domain: userInput });
+}
+```
+
+### Taking a namehash
+
+Namehashing is an algorithm that converts a domain name in a classical format \(like example.crypto\) to a token id that Zilliqa contract can understand. 
+
+{% hint style="warning" %}
+It is important to know the difference between Zilliqa namehashing and [ERC-721](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-137.md#namehash-algorithm) which is a part of [EIP-137](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-137.md).
+In ZIL we use **sha256 from SHA-2**, instead of **keccak256** which is used across Ethereum chain
+{% endhint %}
+
+To do so we need to split the domain by "." character to get each label and then reduce the label's array with a sha256 hashing of an accumulator and next label starting from the end. 
+
+For the purposes of keeping this tutorial short, instead of going into the details of this process, we are going to use the namehash function with some adaptation to the hashing library
+
+{% hint style="info" %}
+Don't forget to add the js-sha256 library to the project in order to use the sha256 function. 
+{% endhint %}
+
+```javascript
+// <!-------> Namehashing functions <!------->
+function namehash(domain) {
+    const parent =
+      '0000000000000000000000000000000000000000000000000000000000000000';
+    return '0x' + [parent]
+      .concat(
+        domain
+          .split('.')
+          .reverse()
+      )
+      .reduce((parent, label) =>
+        childhash(parent, label),
+      );
+  }
+
+function childhash( parent, label) {
+    parent = parent.replace(/^0x/, '');
+    return shaWrapper(parent + shaWrapper(label), "hex");
+  }
+
+function shaWrapper(msg, inputEnc) {
+  if (inputEnc === "hex") {
+    var res = [];
+    msg = msg.replace(/[^a-z0-9]+/ig, '');
+      if (msg.length % 2 !== 0)
+        msg = '0' + msg;
+    for (var i = 0; i < msg.length; i += 2)
+        res.push(parseInt(msg[i] + msg[i + 1], 16));
+    return sha256(res);
+  }
+  return sha256(msg);
+}
```

Comment:
Can we rewrite the whole `namehash` function to be a simple recursive function that returns a `Uint8Array`? (`sha256` can do it.) Then we could introduce a simple function that turns an array into a hex string (like [this](https://stackoverflow.com/a/40031979/4284929)). It would be great if we made it similar to the namehashing algorithm that we reference in the article.

The fact that we convert strings back and forth makes this code really overcomplicated.

Reply:
namehash and childhash functions are taken from our resolution library. shaWrapper is needed because of the sha library we use here. We are not using any bundler and limited to this CDN which doesn't allow us to define the input Encoding. On our resolution side, the shaWrapper is being internal inside the sha function and we don't see it.

I can try to put some comments around it to make it more pleasable, but I think if our users want to know more about namehashing they should go for https://docs.unstoppabledomains.com/domain-registry-essentials/namehashing article. I expect most users reading this guide to either copy the namehashing function into their project or ignore it all at once and write their own algorithm.